Ginger/Ginger/AutomatePageLib/AddActionMenu/SharedRepositoryLib/RepositoryItemUsagePage.xaml.cs

```diff
@@ -314,38 +382,48 @@ private async void UpdateAllButton_Click(object sender, RoutedEventArgs e)
-                }
-
+                });
+                StopProcessingIcon();
+                SetStatus("");
                 Reporter.ToUser(eUserMsgKey.UpdateRepositoryItemUsagesSuccess);
             });
         }
 
         private async void SaveAllBizFlowsButton_Click(object sender, RoutedEventArgs e)
         {
-           await Task.Run(() =>
-           {
-              foreach (RepositoryItemUsage usage in RepoItemUsages)
-               {
-                   if (usage.Status == RepositoryItemUsage.eStatus.Updated ||
-                          usage.Status == RepositoryItemUsage.eStatus.SaveFailed)
-                   {
-                       try
-                       {
-                           Reporter.ToStatus(eStatusMsgKey.SaveItem, null, usage.HostBusinessFlow.Name, GingerDicser.GetTermResValue(eTermResKey.BusinessFlow));
-                           WorkSpace.Instance.SolutionRepository.SaveRepositoryItem(usage.HostBusinessFlow);
-                           usage.Status = RepositoryItemUsage.eStatus.UpdatedAndSaved;
-                           Reporter.HideStatusMessage();
-                       }
-                       catch (Exception ex)
-                       {
-                           usage.Status = RepositoryItemUsage.eStatus.SaveFailed;
-                           Reporter.HideStatusMessage();
-                           Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex);
-                       }
-                   }
-               }
-           }
-            );
+            
+            await Task.Run(() =>
+            {            
+                Parallel.ForEach(RepoItemUsages, usage =>
+                {
+                    StartProcessingIcon();
+                    if (usage.Status == RepositoryItemUsage.eStatus.Updated ||
+                           usage.Status == RepositoryItemUsage.eStatus.SaveFailed)
+                    {
+                        try
+                        {                            
+                            SetStatus("Saving...: " + usage.HostBusinessFlow.Name);
+                            WorkSpace.Instance.SolutionRepository.SaveRepositoryItem(usage.HostBusinessFlow);
+                            usage.Status = RepositoryItemUsage.eStatus.UpdatedAndSaved;
+
+                        }
+                        catch (Exception ex)
+                        {
+                            usage.Status = RepositoryItemUsage.eStatus.SaveFailed;
+
+                            Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex);
+                        }
+                    }
+                });
+            }
+             );
+            StopProcessingIcon();
+            SetStatus("");
         }
+
+        private async void RepositoryItemUsagePageLoaded(object sender, RoutedEventArgs e)
+        {
+            await FindUsages();
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6321030)